dipy/reconst/tests/test_dki_micro.py

```diff
@@ -0,0 +1,217 @@
+multi_params[1, 0, 0] = multi_params[1, 1, 0] = crossing_ref
+
+
+def test_single_fiber_model():
+    # single fiber simulate (which is the assumption of our model)
+    fie = 0.49
+    ADi = 0.00099
+    ADe = 0.00226
+    RDi = 0
+    RDe = 0.00087
+
+    # prepare simulation:
+    theta = random.uniform(0, 180)
+    phi = random.uniform(0, 320)
+    angles = [(theta, phi), (theta, phi)]
+    mevals = np.array([[ADi, RDi, RDi], [ADe, RDe, RDe]])
+    frac = [fie*100, (1 - fie)*100]
+    signal, dt, kt = multi_tensor_dki(gtab_2s, mevals, angles=angles,
+                                      fractions=frac, snr=None)
+    # DKI fit
+    dkiM = dki_micro.DiffusionKurtosisModel(gtab_2s, fit_method="WLS")
+    dkiF = dkiM.fit(signal)
+
+    # Axonal Water Fraction
+    sphere = get_sphere('symmetric724')
+    AWF = dki_micro.axonal_water_fraction(dkiF.model_params, sphere, mask=None,
+                                          gtol=1e-5)
+    assert_almost_equal(AWF, fie)
+
+    # Extra-cellular and intra-cellular components
+    edt, idt = dki_micro.diffusion_components(dkiF.model_params, sphere)
+    EDT = eig_from_lo_tri(edt)
+    IDT = eig_from_lo_tri(idt)
+
+    # check eigenvalues
+    assert_array_almost_equal(EDT[0:3], np.array([ADe, RDe, RDe]))
+    assert_array_almost_equal(IDT[0:3], np.array([ADi, RDi, RDi]))
+    # first eigenvalue should be the direction of the fibers
+    fiber_direction = _check_directions([(theta, phi)])
+    f_norm = abs(np.dot(fiber_direction, np.array((EDT[3], EDT[6], EDT[9]))))
+    assert_almost_equal(f_norm, 1.)
+    f_norm = abs(np.dot(fiber_direction, np.array((IDT[3], IDT[6], IDT[9]))))
+    assert_almost_equal(f_norm, 1.)
+
+    # Test model and fit objects
+    wmtiM = dki_micro.KurtosisMicrostructuralModel(gtab_2s, fit_method="WLS")
+    wmtiF = wmtiM.fit(signal)
+    assert_almost_equal(wmtiF.awf, AWF)
+    assert_almost_equal(wmtiF.axonal_diffusivity, ADi)
+    assert_array_almost_equal(wmtiF.hindered_evals,
+                              np.array([ADe, RDe, RDe]))
+    assert_array_almost_equal(wmtiF.restricted_evals,
+                              np.array([ADi, RDi, RDi]))
+    assert_almost_equal(wmtiF.hindered_ad, ADe)
+    assert_almost_equal(wmtiF.hindered_rd, RDe)
+    assert_almost_equal(wmtiF.tortuosity, ADe/RDe)
+
+
+def test_wmti_model_multi_voxel():
+    # single fiber simulate (which is the assumption of our model)
+    FIE = np.array([[[0.30, 0.32], [0.74, 0.51]],
+                    [[0.47, 0.21], [0.80, 0.63]]])
+    RDI = np.zeros((2, 2, 2))
+    ADI = np.array([[[1e-3, 1.3e-3], [0.8e-3, 1e-3]],
+                    [[0.9e-3, 0.99e-3], [0.89e-3, 1.1e-3]]])
+    ADE = np.array([[[2.2e-3, 2.3e-3], [2.8e-3, 2.1e-3]],
+                    [[1.9e-3, 2.5e-3], [1.89e-3, 2.1e-3]]])
+    Tor = np.array([[[2.6, 2.4], [2.8, 2.1]],
+                    [[2.9, 2.5], [2.7, 2.3]]])
+    RDE = ADE / Tor
+
+    # prepare simulation:
+    DWIsim = np.zeros((2., 2., 2., gtab_2s.bvals.size))
```

Comment:
These all need to be integers! 

    DWIsim = np.zeros((2, 2, 2, gtab_2s.bvals.size))